src/mappings/SchemeFactory/mapping.ts

```diff
@@ -0,0 +1,101 @@
+import { Address, BigInt, Bytes } from '@graphprotocol/graph-ts';
+
+// Import event types from the Reputation contract ABI
+import {
+  NewSchemeProposal,
+  ProposalExecuted,
+} from '../../types/SchemeFactory/SchemeFactory';
+
+import * as domain from '../../domain';
+
+// Import entity types generated from the GraphQL schema
+import {
+   SchemeFactoryNewSchemeProposal,
+   SchemeFactoryProposal,
+   SchemeFactoryProposalExecuted,
+} from '../../types/schema';
+import { equalsBytes, equalStrings, eventId } from '../../utils';
+
+export function handleNewSchemeProposal(event: NewSchemeProposal): void {
+  let ent = SchemeFactoryNewSchemeProposal.load(eventId(event));
```

Comment:
how about the id will be the proposalId?

Reply:
That's how we did it in the SchemeRegistrar and other schemes. We can change it I guess but didn't want to make it" stand out".

Reply:
well:
1. event is unique and for that this line is redundant if we use event
2. I feel that proposalId is better

Reply:
ok...